server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -520,6 +528,16 @@ public String updateConfiguration(final long userId, final String name, final St
                     _accountDetailsDao.update(accountDetailVO.getId(), accountDetailVO);
                 }
                 break;
+
+            case ImageStore:
+                final ImageStoreVO imgStore = _imageStoreDao.findById(resourceId);
+                if (imgStore == null) {
+                    throw new InvalidParameterValueException("unable to find image store by id " + resourceId);
```

Comment:
``InvalidParameterValueException`` is used to indicate that an invalid parameter was passed by a caller to a method.  Since ``imgStore`` was not passed into this method, it does not seem appropriate to use this exception type.  ``IllegalStateException`` seems more appropriate.  Also, consider using ``Preconditions.checkState`` to replace the ``if`` block with a single line of code for greater concession.